instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_server.py

```diff
@@ -136,74 +189,57 @@ def get_from_grpc_metadata(metadata, key) -> List[str]:
         else:
             yield
 
-    def _start_span(self, method):
-        span = self._tracer.start_as_current_span(
-            name=method, kind=trace.SpanKind.SERVER
+    def _start_span(self, handler_call_details, context):
+
+        attributes = {
+            "grpc.method": handler_call_details.method,
+        }
+
+        metadata = dict(context.invocation_metadata())
+        if "user-agent" in metadata:
+            attributes["grpc.user_agent"] = metadata["user-agent"]
+
+        # Split up the peer to keep with how other telemetry sources
+        # do it.  This looks like ipv6:[::1]:57284 or ipv4:127.0.0.1:57284.
+        host, port = context.peer().split(":", 1)[1].rsplit(":", 1)
```

Comment:
Is this guaranteed to b the same format?

Reply:
As far as I can tell, but I could put a check in there to be sure.

Reply:
I found some references to expected formats in the C library, so it looks like yes, this is the format it will be.... except for that one which is comma-separated, so I definitely need to update this.  Good catch!

https://github.com/grpc/grpc/blob/master/test/core/client_channel/resolvers/sockaddr_resolver_test.cc#L90

Reply:
Sounds good. I'd suggest to be defensive around code like this and perhaps wrap it in a try .. except block with an error log. We don't want instrumentation to break apps :)

Reply:
I updated it and made the log a warning rather than error, since it's not really an error.

Reply:
Didn't read it. I'd say it actually is an error as an exception is raised. We are just silencing the error so it doesn't crash users' apps. IMO it should surface in the loudest possible manner without crashing apps so we get to know about it and fix the instrumentation. WDYT?

Reply:
I don't agree.  This condition is not something I as the application developer would have any control over, and by making it an error rather than warning, now I have to consider suppressing messages from the tracing library in order to not have messages about errors I can't fix in my logs, which I may do things like trigger alerts with via tools such as sentry.io.

Reply:
Fair enough.